pan/functions.pan

```diff
@@ -1,270 +1,290 @@
+    "/data" = dict("one", 1);
+    "/data" = npush("two", 2, "three", 3);
+
+    "/data" will contain dict("one", 1, "two", 2, "three", 3)
+}
 function npush = {
     # check the cardinality of the arguments; must be a
     # multiple of two
     if ((ARGC % 2) != 0) {
-        error("usage: \"/full/path\" = npush(key, value, ...)");
-    };
-
-    # Determine what initial value to use.
-    if (exists(SELF) && is_nlist(SELF)) {
-        v = SELF;
-    } else if (!exists(SELF) || !is_defined(SELF)) {
-        v = nlist();
-    } else {
-        error("npush can only be applied to a nlist");
+        error("usage: npush(key, value [, key1, value1, ...]) requires an even number of arguments");
     };
 
     # Create a new nlist from the given values.
     n = nlist();
     index = 0;
     while (index < ARGC) {
-        if (!is_string(ARGV[index]))
-        error("key is not a string: "+to_string(ARGV[index]));
+        if (!is_string(ARGV[index])) {
+            error("key is not a string: "+to_string(ARGV[index]));
+        };
         n[ARGV[index]] = ARGV[index+1];
         index = index + 2;
     };
 
-    # Returned the merged values.
-    return(merge(v,n));
+    # Determine what initial value to use.
+    if (exists(SELF) && is_nlist(SELF)) {
+        merge(SELF, n);
+    } else if (!exists(SELF) || !is_defined(SELF)) {
+        n;
+    } else {
+        error("npush can only be applied to a nlist");
+    };
 };
 
 
-############################################################
-##=
-## @function push_if
-## @# pushes zero or more values onto the end of a list
-##+if the flag is true.  If the flag is false, then either
-##+"SELF" (if it is defined) or an empty list is returned.
-## @syntax flag:boolean values:element
-## @param:flag flag to determine if value is added
-## @param:value... value(s) to add
-## @example
-##+# "/data" will contain list(1,2,3)
-##+variable flag = true;
-##+"/data" = list(1);
-##+"/data" = push_if(flag,2,3);
-##=
-############################################################
+@documentation{
+    pushes zero or more values onto the end of a list
+    if the flag is true.  If the flag is false, then either
+    "SELF" (if it is defined) or an empty list is returned.
+    @syntax flag:boolean values:element
+    @param:flag flag to determine if value is added
+    @param:value... value(s) to add
+
+    Example:
+    variable CONDITION = true;
+    "/data" = list(1);
+    "/data" = push_if(CONDITION, 2, 3);
+
+    "/data" will contain list(1, 2, 3)
+}
 function push_if = {
     # Check cardinality and first argument.
-    if (ARGC == 0 || !is_boolean(ARGV[0]))
-        error("usage: \"/full/path\" = push(boolean, values...)");
+    if (ARGC == 0 || !is_boolean(ARGV[0])) {
+        error("usage: push_if(boolean, values...) requires at least a boolean as first argument");
+    };
 
     if (ARGV[0]) {
-        # Get the reference to SELF or create an empty list
-        # as necessary.
+        delete(ARGV[0]);
```

Comment:
because it is the boolean condition, not a value to push (old code started looping from 1)